immutable.d.ts

```diff
@@ -21,6 +21,20 @@ interface KeyValuePair<K, V> extends Array<K | V> {
   readonly 1: V;
 }
 
+type Prev = [never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, ...0[]]
```

Comment:
what happens if index is greater than `15`?

Reply:
If index is greater than 15 you have some extreme nesting, i.e.
`myObj.a.b.c.d.e.f.g.h.i.j.k.l.m.n.o` 

I'm pretty sure more than 15 nested values would break this type, in which case you'd have to typecast to tell TypeScript what your type is, but I don't see this as a problem considering the current solution requires that for any length (since it doesn't try to type anything dot-nested)

Reply:
As for the second part, I'm not really following.. This project doesn't have any `.ts` files other than `immutable.d.ts`, right? Would you like me to add a unit test file? If so, in `src` with the `.js` files, or outside of `src` with `immutable.d.ts`?

Reply:
It also appears there's no testing framework associated with this lib?

Reply:
<img width="1083" alt="Screen Shot 2022-07-12 at 2 09 11 PM" src="https://user-images.githubusercontent.com/12897328/178563190-b0ce44a7-043f-4545-ab9e-59e6eee6153e.png">

Reply:
Added some type testing as well!

Reply:
thanks for the comments - will proceed further with moving this PR into `master`